pandas/tests/io/excel/test_readers.py

```diff
@@ -870,6 +870,27 @@ def test_excel_passes_na(self, read_ext):
         )
         tm.assert_frame_equal(parsed, expected)
 
+    @pytest.mark.parametrize("na_filter", [None, True, False])
+    def test_excel_passes_na_filter(self, read_ext, na_filter):
+        # gh-25453
+        kwargs = {}
+
+        if na_filter is not None:
+            kwargs["na_filter"] = na_filter
+
+        with pd.ExcelFile("test5" + read_ext) as excel:
+            parsed = pd.read_excel(
+                excel, "Sheet1", keep_default_na=True, na_values=["apple"], **kwargs
```

Comment:
Looks good. Do we already have coverage for the `keep_default_na=False` case documented above?

Reply:
Which case are you talking about?  The documentation mentions multiple `keep_default_na=False` cases.

Reply:
Right was asking if we already have coverage on `keep_default_na=False` since mentioned in docs, but only True here (in contrast to `na_filter`)

Reply:
The test above has coverage for `keep_default_na=False`.